code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/unit/DefaultDeploymentUnitInstance.java

```diff
@@ -229,14 +232,18 @@ public boolean isHealthCheckInitializing() {
 
     @Override
     public void waitForAllocate() {
-        if (this.instance != null) {
-            instance = context.resourceMonitor.waitFor(instance, new ResourcePredicate<Instance>() {
-                @Override
-                public boolean evaluate(Instance obj) {
-                    return context.objectManager.find(InstanceHostMap.class, INSTANCE_HOST_MAP.INSTANCE_ID,
-                            instance.getId()).size() > 0;
-                }
-            });
+        try {
+            if (this.instance != null) {
+                instance = context.resourceMonitor.waitFor(instance, new ResourcePredicate<Instance>() {
+                    @Override
+                    public boolean evaluate(Instance obj) {
+                        return context.objectManager.find(InstanceHostMap.class, INSTANCE_HOST_MAP.INSTANCE_ID,
+                                instance.getId()).size() > 0;
+                    }
+                });
+            }
+        } catch (Exception ex) {
+            throw new ServiceInstanceAllocateException("Failed to allocate instance [" + instance.getId() + "]");
```

Comment:
Probably don't want to mask the original exception.  Can you pass it to the constructor?